server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java

```diff
@@ -12,17 +12,24 @@
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
+import java.util.List;
 
-public record HealthIndicatorResult(String name, String component, HealthStatus status, String summary, HealthIndicatorDetails details)
-    implements
-        ToXContentObject {
+public record HealthIndicatorResult(
+    String name,
+    String component,
+    HealthStatus status,
+    String summary,
+    HealthIndicatorDetails details,
+    List<HealthIndicatorImpact> impacts
+) implements ToXContentObject {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         builder.field("status", status.xContentValue());
         builder.field("summary", summary);
         builder.field("details", details, params);
+        builder.field("impacts", impacts);
```

Comment:
I think we can skip returning impacts if none are reported?

Reply:
Will taking away fields complicate things for the API consumers? I guess they must have a way to deal with optional fields in the response. I'll remove it and we can put it back if needed.